### PR TITLE
Setting up server API

### PR DESCRIPTION
**Pull request by jpodivin**
The server now has properly defined API, with data model of response. Error handling was implemented for various failure modes, including possibility of invalid log probabilities.

### CODE REVIEW

File: logdetective/server.py [hunk 1]
```diff
@@ -1,21 +1,29 @@
...
 class BuildLog(BaseModel):
     """Model of data submitted to API.
     """
     url: str
 
+
+class Response(BaseModel):
+    """Model of data returned by Log Detective API
+    """
+    llm_response: CreateCompletionResponse
```
Comment (TomasTomecek) [inline logdetective/server.py:23]: Can we see an example content of this?

Comment (jpodivin) [reply logdetective/server.py:23]: Sure, it's all part of llama-cpp-python API https://llama-cpp-python.readthedocs.io/en/latest/api-reference/#llama_cpp.llama_types.CreateCompletionResponse

Comment (TomasTomecek) [reply logdetective/server.py:23]: nice, that's pretty detailed

please link it in the code so it's easy to find :)

**Review by TomasTomecek — APPROVED** [review]
very nice, we're getting there with the service \o/

File: logdetective/server.py (hunk 1 above)
Comment (jpodivin) [reply logdetective/server.py:27]: Done

**Pull request merged**